rustworkx-core/src/generators/random_graph.rs

```diff
@@ -856,17 +860,23 @@ where
 }
 
 #[inline]
-fn hyperbolic_distance(p1: &[f64], p2: &[f64]) -> f64 {
-    if p1.iter().chain(p2.iter()).any(|x| x.is_infinite()) {
-        f64::INFINITY
+fn hyperbolic_distance(x: &[f64], y: &[f64]) -> f64 {
+    let mut sum_squared_x = 0.;
+    let mut sum_squared_y = 0.;
+    let mut inner_product = 0.;
+    for (x_i, y_i) in x.iter().zip(y.iter()) {
+        if x_i.is_infinite() || y_i.is_infinite() || x_i.is_nan() || y_i.is_nan() {
+            return f64::NAN;
+        }
+        sum_squared_x += x_i * x_i;
+        sum_squared_y += y_i * y_i;
+        inner_product += x_i * y_i;
```

Comment:
```suggestion
        sum_squared_x = x_i::mul_add(x_i, x_i, sum_squared_x);
        sum_squared_y = x_i::mul_add(y_i, y_i, sum_squared_y);
        inner_product = x_i::mul_add(x_i, y_i, inner_product);
```
I think we might as well use fused add-multiply after we confirmed the floats are valid

Reply:
I wasn't aware of that function, good to know!